### DISCUSSION TITLE
Cannot create record with prisma and mongodb locally

### DISCUSSION THREAD

**Discussion by RichardBray** [prisma/prisma#10128] [Q&A] [answered]
Hey all,
I've been messing about with Prisma, Express and Graphql. I'm connecting to a local instance of MongoDB with this .env variable:
```
DATABASE_URL="mongodb://<username>:<password>@localhost:27017/test?readPreference=primary&authSource=admin&appname=MongoDB%20Compass&directConnection=true&ssl=false&retryWrites=false
```

This works fine in MongoDB Compass but I get this error whenever I try to create a record with Prisma:

```json5
{
  "errors": [
    {
      "message": "\nInvalid `prisma.todos.create()` invocation:\n\n\n  Error occurred during query
 execution:\nConnectorError(ConnectorError { user_facing_error: None, kind: RawError { code: \"unknown\", 
message: \"Command failed (IllegalOperation): This MongoDB deployment does not support retryable writes. 
Please add retryWrites=false to your connection string.)\" } })",
...
    }
  ],
}
```

Even after adding `retryWrites=false` I still get the above error. 

I am able to create records in my Atlas cluster with Prisma but not with my local MongoDB instance.

Any ideas?

**Comment by RichardBray** [comment]
My issue is similar to the issue raised here:
https://github.com/prisma/studio/issues/771

But it's not specific to Prisma Studio.

**Comment by austincrim** [comment, marked as answer]
Hey @RichardBray 👋,

Prisma Client requires a MongoDB replica set. We note this constraint in our [MongoDB guides](https://www.prisma.io/docs/getting-started/setup-prisma/start-from-scratch/mongodb-typescript-mongodb#prerequisites) under "Prerequisites". It is possible to [set up replica sets locally](https://docs.mongodb.com/manual/tutorial/deploy-replica-set/), but Mongo Atlas is definitely the easiest way to get started.

Let me know if that helps!

**Reply by RichardBray** [reply]
Cool, yeah I've been messing about with trying to set up a replica set locally, went through a bunch of resources like the ones listed below:
- https://gist.github.com/davisford/bb37079900888c44d2bbcb2c52a5d6e8
- https://www.thomaszhang.com/mongodb-cluster-administration-part2
- https://docs.mongodb.com/manual/tutorial/convert-standalone-to-replica-set/

But it's honestly super-tough 😢 
I think I'll stick to Atlas for now.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
